### PR TITLE
Scope pretrained weights loading

### PR DESCRIPTION
**Pull request by Laughing-q**
@glenn-jocher this PR scope pretrained weights loading to `model.train` from `cfg.entrypoint`, this way makes more sense to me as `pretrained` is used for train mode only if I remember this correctly. 
And also it'd allow following usage work as expected:
```python
def test_pretrained_loading():
    from ultralytics import YOLO

    # NOTE: test the pretrained=yolo11n.pt works properly
    model = YOLO("yolo11n.yaml")
    model.train(data="coco128.yaml", epochs=2, pretrained="yolo11n.pt", verbose=False)

test_pretrained_loading()
```
- Using main branch it'd not load pretrained weight as expected.
- It works as expected with this PR.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Improved how pretrained model weights are loaded during training for a smoother user experience. 🚀

### 📊 Key Changes
- Moved the logic for loading pretrained weights from the config entrypoint to the model's training method.
- Now, if a pretrained model path is provided during training, the model automatically loads those weights.

### 🎯 Purpose & Impact
- Ensures pretrained weights are always loaded at the right time, reducing potential errors and confusion.
- Makes the training process more intuitive for users who want to start from existing weights.
- Enhances reliability and consistency when customizing or resuming training with pretrained models.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Laughing-q, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response to help you streamline the review process—an Ultralytics engineer will also provide feedback soon.

To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Make sure your commit messages are clear, concise, and follow the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is up to date with the `ultralytics/ultralytics` `main` branch. If your branch is behind, update it by clicking the 'Update branch' button or running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks pass. If any checks fail, please address them.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) if you have introduced or modified features.
- ✅ **Add Tests**: If applicable, add or update tests to cover your changes, and confirm all tests are passing.
- ✅ **Sign the CLA**: If this is your first PR to Ultralytics, please sign our [Contributor License Agreement (CLA)](https://docs.ultralytics.com/help/CLA/) by adding a comment with: "I have read the CLA Document and I sign the CLA".
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more details, see our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). If you have any questions, feel free to leave a comment. Thank you for helping improve Ultralytics! 🌟

**Review by ambitious-octopus — APPROVED** [review]
@Laughing-q nice!

**Comment by UltralyticsAssistant** [comment]
🎉 Fantastic work, @Laughing-q and @ambitious-octopus—your PR is now merged! 

As Henry Ford once said, "Coming together is a beginning; keeping together is progress; working together is success." Your collaborative effort has truly made the user experience on Ultralytics HUB even smoother by making pretrained model loading smarter and more intuitive. This enhancement will empower our community to train and customize models with greater confidence and ease.

Thank you both for your dedication and meaningful contributions! 🚀
